magento/magento-cloud-patches
# MCLOUD-6423: MC-34580: Magento Quality Patches

### Description
https://jira.corp.magento.com/browse/MCLOUD-6423
Story [Magento Quality Patches](https://jira.corp.magento.com/browse/MC-34580)


### Preconditions for test scenarios
- Check if your GitHub token has access to https://github.com/magento-mpi/magento-cloud-patches.git, https://github.com/magento/quality-patches.git, https://github.com/magento-mpi/ece-tools.git
- Add to composer.json 

```
 "repositories": {
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        },
        "mcp": {
            "type": "vcs",
            "url": "https://github.com/magento-mpi/magento-cloud-patches.git"
        },
        "mqp": {
            "type": "vcs",
            "url": "https://github.com/magento/quality-patches.git"
        },
        "ece": {
            "type": "vcs",
            "url": "https://github.com/magento-mpi/ece-tools.git"
        }
    },
    "require": {
        "magento/magento-cloud-patches": "dev-MC-34580 as 1.0.99",
        "magento/quality-patches": "dev-master as 1.0.999",
        "magento/ece-tools": "dev-MC-34668 as 2002.1.99"
    },
```
### Testing scenario 1

- Add to .magento.env.yaml
```
 stage:
    build:
        QUALITY_PATCHES:
            - MCTEST-1001
```

- Install Magento
- Check /var/log/patch.log file contains 
```
INFO: Patch MCTEST-1001 has been applied {"file":"/app/vendor/magento/quality-patches/src/../patches/MCTEST-1001__TEST_changes__2.3.5_ce.patch"}
INFO: Patch MCTEST-1001 has been applied {"file":"/app/vendor/magento/quality-patches/src/../patches/MCTEST-1001__TEST_changes__2.3.5_ee.patch"}
```
- Verify that patches https://github.com/magento/quality-patches/blob/master/patches/MCTEST-1001__TEST_changes__2.3.5_ce.patch and https://github.com/magento/quality-patches/blob/master/patches/MCTEST-1001__TEST_changes__2.3.5_ee.patch were applied (check sources in vendor folder) 
- In console run './vendor/bin/ece-patches status' command and verify that all 'Required' patches have 'Applied' status in a table
- Check that 'Optional' patch  MCTEST-1001 has 'Applied' status and MCTEST-1001,  MCTEST-1002 have 'Not applied' statuses

### Testing scenario 2
- Remove QUALITY_PATCHES from .magento.env.yaml
- Install Magento
- In console run './vendor/bin/ece-patches status' command and verify that all 'Required' patches have 'Applied' status in a table
- Check that all 'Optional' patches  have  'Not Applied' status

### Testing scenario 3
- Add to .magento.env.yaml
```
 stage:
    build:
        QUALITY_PATCHES:
            - MCTEST-1003
```
- Install Magento
- Check that 'Optional' patch  MCTEST-1003 has 'Applied' status and MCTEST-1002, MCTEST-1001  have 'Applied' status as well since MCTEST-1003 requires MCTEST-1002 and MCTEST-1002 requires MCTEST-1001

### Testing scenario 4
- Using the same Magento instance run './vendor/bin/ece-patches revert MCTEST-1003'
- Check error message 'Revert command is unavailable on Magento Cloud' (revert command should not be available on Cloud)

### Contribution checklist
